### PR TITLE
Making it buildable with jdk8 and fixed findbugs issues

### PR DESCRIPTION
**Pull request by rsandell**
The change could only require 1.580.1 but then I wouldn't get to use the wonderful ```Jenkins.getActiveInstance()```

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
:bug: . It bumps the core dependency and makes to LTS lines unsupported without a particular need.
Let's bump it to the nearest LTS, which is 1.580.x

**Comment by rsandell** [comment]
As I wrote getActiveInstance() is not in 1.580 and would require a larger change.

**Comment by oleg-nenashev** [comment]
Probably. But this justification is not enough, because it does not bring user-visible fixes/features to the plugin. You can easily implement a utility method doing the same thing

**Comment by rsandell** [comment]
Hmm, tests are now failing in jdk7, something that I haven't seen when I've done similar operations on other plugins. Need to investigate.

**Comment by rsandell** [comment]
Even more hmm, test are working locally for me on the same jdk 1.7.0_79, I'm using a newer maven (3.3.3) though, but the error doesn't hint that that would be the problem.

**Comment by rsandell** [comment]
@stephenc or @jglick do you have any insight?

**Comment by jglick** [comment]
Dunno, some error was thrown during class initialization and swallowed somewhere, and then the JRE fails to report it as a `cause` in subsequent `NoClassDefFoundError`s.

File: src/main/java/hudson/cli/MailCommand.java [hunk 1]
```diff
@@ -46,8 +46,13 @@ public String getShortDescription() {
...
-        Jenkins.getInstance().checkPermission(Item.CONFIGURE);
-        Transport.send(new MimeMessage(Mailer.descriptor().createSession(),stdin));
-        return 0;
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            jenkins.checkPermission(Item.CONFIGURE);
+            Transport.send(new MimeMessage(Mailer.descriptor().createSession(),stdin));
+            return 0;
+        } else {
+            throw new IllegalStateException("Jenkins is not active.");
```
Comment (jglick) [inline src/main/java/hudson/cli/MailCommand.java]: :ant: code like this should include

```java
// TODO 1.590+ Jenkins.getActiveInstance
```

**Comment by jglick** [comment]
:bee: assuming the test failures get sorted out.

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by rsandell** [comment]
@reviewbybees done

**Pull request merged**